### PR TITLE
Add lyrics translation using Microsoft Bing API

### PR DESCRIPTION
**Pull request by Kraymer**
:globe_with_meridians: Adiou,  

Dunno if it should be merged or if I should maintain it separately in my fork : scratching my own hitch here, and the result of translation can be *suboptimal*.

- [x] fix pep8 warnings on existing code
- [X] add config option to restrict the translation feature to few langages
- [X] documentation  
- [x]  detect when token expired and get a new one
- [x]   use a ``set`` to reduce size of text to translate and thus translate more songs within the fixed API limit of 2,000,000 chars/month

---

    lyrics: fetched lyrics: João Gilberto - Amoroso/Brasil - Aquarela do Brasil (Watercolour Painting of Brazil)
    Brasil! / Brazil! 
    Meu Brasil Brasileiro /  My Brazilian Brazil 
    Meu mulato inzoneiro /  My creamy inzoneiro 
    Vou cantar-te nos meus versos /  I will sing you my verses 
    Brasil, samba que dá /  Brazil, samba gives 
    Bamboleio, que faz gingar /  Wiggle, which makes waddle 
    O Brasil do meu amor /  The Brazil of my love 
    Terra de Nosso Senhor... /  Land of our Lord ... 

[...]

### CODE REVIEW

File: docs/plugins/lyrics.rst [hunk 1]
```diff
@@ -113,3 +121,19 @@ After that, the lyrics plugin will fall back on other declared data sources.
 
 .. _pip: http://www.pip-installer.org/
 .. _BeautifulSoup: http://www.crummy.com/software/BeautifulSoup/bs4/doc/
+
+Activate on-the-fly translation
```
Comment (sampsyo) [inline docs/plugins/lyrics.rst]: Matching our usual style, this should probably be in Title Case: "Activate On-the-Fly Translation"

File: docs/plugins/lyrics.rst [hunk 2]
```diff
@@ -38,6 +38,14 @@ configuration file. The available options are:
 
 - **auto**: Fetch lyrics automatically during import.
   Default: ``yes``.
+- **bing_client_secret**: Your Bing Translation application password (to enable
+  on-the-fly translating)
```
Comment (sampsyo) [inline docs/plugins/lyrics.rst]: This should probably link to the section that explains more, below.

File: docs/plugins/lyrics.rst [hunk 3]
```diff
@@ -113,3 +121,19 @@ After that, the lyrics plugin will fall back on other declared data sources.
...
+
+Using the Bing Translation API requires `langdetect`_, which you can install
+using `pip`_ by typing::
+
+    pip install langdetect
+
+You also need to register for a Microsoft Azure Marketplace free account and
+to the `Microsoft Translator API`_.
+Set the ``bing_client_secret`` configuration option to your registered
+application password, alongside ``bing_lang_to`` target `language code`_.
```
Comment (sampsyo) [inline docs/plugins/lyrics.rst]: Perhaps we should include a few more instructions here about exactly how to get a client secret?

File: beetsplug/lyrics.py [hunk 4]
```diff
@@ -671,3 +703,24 @@ def get_lyrics(self, artist, title):
...
+        if not self.bing_auth_token:
+            self.bing_auth_token = self.get_bing_access_token()
+        if self.bing_auth_token:
+            headers = {"Authorization ": self.bing_auth_token}
+            url = ('http://api.microsofttranslator.com/v2/Http.svc/'
+                   'Translate?text=%s&to=%s' % (text.replace('\n', '|'),
+                                                to_language))
+            r = requests.get(url, headers=headers)
+            if r.status_code != 200:
+                self._log.debug(r.text)
```
Comment (sampsyo) [inline beetsplug/lyrics.py]: This should probably be a little more explicit, like:

    self._log.debug('translation API error {}: {}', r.status_code, r.text)

File: beetsplug/lyrics.py [hunk 5]
```diff
@@ -16,17 +16,18 @@
...
 import re
 import requests
-import json
 import unicodedata
 import urllib
-import difflib
-import itertools
 import warnings
 from HTMLParser import HTMLParseError
+from langdetect import detect
```
Comment (sampsyo) [inline beetsplug/lyrics.py]: This should be conditionally imported, so the library isn't required if you don't enable translation.

**Comment by sampsyo** [comment]
This is pretty nifty! I'm all for it—it's a fairly "niche" feature, but that's true of a great many of beets' features. :smiley:

File: docs/plugins/lyrics.rst (hunk 1 above)
Comment (Kraymer) [inline docs/plugins/lyrics.rst]: :white_check_mark:

File: docs/plugins/lyrics.rst (hunk 2 above)
Comment (Kraymer) [inline docs/plugins/lyrics.rst]: :white_check_mark:

File: docs/plugins/lyrics.rst (hunk 3 above)
Comment (Kraymer) [inline docs/plugins/lyrics.rst]: :white_check_mark:

File: beetsplug/lyrics.py (hunk 4 above)
Comment (Kraymer) [inline beetsplug/lyrics.py]: :white_check_mark:

**Comment by Kraymer** [comment]
:ghost:  Good to merge?

**Comment by sampsyo** [comment]
Looks great! Go for it!

**Pull request merged**